WordPress/Classes/ViewRelated/Blog/CreateNewBlog/CreateNewBlogViewController.m

```diff
@@ -0,0 +1,551 @@
+        _siteTitleField.autocorrectionType = UITextAutocorrectionTypeNo;
+        _siteTitleField.autocapitalizationType = UITextAutocapitalizationTypeNone;
+        _siteTitleField.autoresizingMask = UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleLeftMargin;
+        _siteTitleField.accessibilityIdentifier = @"Title";
+        _siteTitleField.returnKeyType = UIReturnKeyNext;
+        [self.view addSubview:_siteTitleField];
+    }
+    
+    // Add Site Address
+    if (_siteAddressField == nil) {
+        _siteAddressField = [[WPWalkthroughTextField alloc] initWithLeftViewImage:[UIImage imageNamed:@"icon-url-field"]];
+        _siteAddressField.backgroundColor = [UIColor whiteColor];
+        _siteAddressField.placeholder = NSLocalizedString(@"Blog Address", nil);
+        _siteAddressField.font = [WPNUXUtility textFieldFont];
+        _siteAddressField.adjustsFontSizeToFitWidth = YES;
+        _siteAddressField.delegate = self;
+        _siteAddressField.autocorrectionType = UITextAutocorrectionTypeNo;
+        _siteAddressField.autocapitalizationType = UITextAutocapitalizationTypeNone;
+        _siteAddressField.returnKeyType = UIReturnKeyDone;
+        _siteAddressField.showTopLineSeparator = YES;
+        _siteAddressField.autoresizingMask = UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleLeftMargin;
+        _siteAddressField.accessibilityIdentifier = @"Blog Address";
+        [self.view addSubview:_siteAddressField];
+        
+        // add .wordpress.com label to textfield
+        _siteAddressWPComLabel = [[UILabel alloc] init];
+        _siteAddressWPComLabel.text = @".wordpress.com";
+        _siteAddressWPComLabel.textAlignment = NSTextAlignmentCenter;
+        _siteAddressWPComLabel.font = [WPNUXUtility descriptionTextFont];
+        _siteAddressWPComLabel.textColor = [WPStyleGuide allTAllShadeGrey];
+        [_siteAddressWPComLabel sizeToFit];
+        
+        UIEdgeInsets siteAddressTextInsets = [(WPWalkthroughTextField *)_siteAddressField textInsets];
+        siteAddressTextInsets.right += _siteAddressWPComLabel.frame.size.width + 10;
+        [(WPWalkthroughTextField *)_siteAddressField setTextInsets:siteAddressTextInsets];
+        [_siteAddressField addSubview:_siteAddressWPComLabel];
+    }
+    
+    // Add Next Button
+    if (_createBlogButton == nil) {
+        _createBlogButton = [[WPNUXMainButton alloc] init];
+        [_createBlogButton setTitle:NSLocalizedString(@"Create WordPress.com blog", nil) forState:UIControlStateNormal];
+        _createBlogButton.enabled = NO;
+        [_createBlogButton addTarget:self action:@selector(createBlogButtonAction) forControlEvents:UIControlEventTouchUpInside];
+        [_createBlogButton sizeToFit];
+        _createBlogButton.autoresizingMask = UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleLeftMargin;
+        [self.view addSubview:_createBlogButton];
+    }
+}
+
+- (void)layoutControls
```

Comment:
@kwonye this is technically called twice via `viewDidLoad initializeView` and also `viewWillAppear layoutControls`. Might need a rework to something along the lines of `layoutControlsIfNeeded`